pkg/hooks/loader.go

```diff
@@ -225,14 +246,75 @@ func (h *Hook) GetConfigMocks() ([]*models.Mock, error) {
 
 func (h *Hook) DeleteTcsMock(mock *models.Mock) bool {
 	isDeleted := h.tcsMocks.delete(mock.TestModeInfo)
+	if isDeleted {
+		h.UpdatePersistentMatchedMock(mock.Name, true)
+	}
 	return isDeleted
 }
 
 func (h *Hook) DeleteConfigMock(mock *models.Mock) bool {
 	isDeleted := h.configMocks.delete(mock.TestModeInfo)
+	if isDeleted {
+		h.UpdatePersistentMatchedMock(mock.Name, false)
+	}
 	return isDeleted
 }
 
+func (h *Hook) GetUsedMocks(testSet string) ([]*models.Mock, error) {
+	var matchedMocks []*models.Mock
+	tcsMocks, err := h.TestCaseDB.ReadTcsMocks(nil, testSet)
+	if err != nil {
+		return nil, err
+	}
+	configMocks, err := h.TestCaseDB.ReadConfigMocks(testSet)
+	if err != nil {
+		return nil, err
+	}
+	TcsMocks := []*models.Mock{}
+	for _, mock := range tcsMocks {
+		tcsmock, ok := mock.(*models.Mock)
+		if !ok {
+			continue
+		}
+		TcsMocks = append(TcsMocks, tcsmock)
+	}
+	ConfigMocks := []*models.Mock{}
+	for _, mock := range configMocks {
+		configMock, ok := mock.(*models.Mock)
+		if !ok {
+			continue
+		}
+		ConfigMocks = append(ConfigMocks, configMock)
+	}
+	mocks := append(ConfigMocks, TcsMocks...)
+	for _, mock := range mocks {
+		if _, ok := h.persistentMatchedMocks[mock.Name]; ok {
+			matchedMocks = append(matchedMocks, mock)
+		}
+	}
+	return matchedMocks, nil
```

Comment:
there are config calls that go in between like health check or etc for that sake its better to sort all mocks on name itself